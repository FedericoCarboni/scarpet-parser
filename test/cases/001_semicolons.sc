// Optional trailing semicolons test
(
    null;
    null;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
);

f(
    null;,
    null;,
);
