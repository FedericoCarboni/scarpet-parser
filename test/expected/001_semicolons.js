/** @type {import('../../lib/Parser.js').Node} */
export const root = {
    kind: 'BinaryExpression',
    operator: ';',
    lvalue: {
        kind: 'BinaryExpression',
        operator: ';',
        lvalue: {
            kind: 'ParenthesisedExpression',
            value: {
                kind: 'BinaryExpression',
                operator: ';',
                lvalue: {
                    kind: 'BinaryExpression',
                    operator: ';',
                    lvalue: {
                        kind: 'BinaryExpression',
                        operator: ';',
                        lvalue: {
                            kind: 'BinaryExpression',
                            operator: ';',
                            lvalue: {
                                kind: 'Constant',
                                name: 'null',
                                start: {
                                    offset: 43,
                                    line: 2,
                                    character: 4,
                                },
                                end: {
                                    offset: 47,
                                    line: 2,
                                    character: 8,
                                },
                            },
                            rvalue: {
                                kind: 'Constant',
                                name: 'null',
                                start: {
                                    offset: 53,
                                    line: 3,
                                    character: 4,
                                },
                                end: {
                                    offset: 57,
                                    line: 3,
                                    character: 8,
                                },
                            },
                            comment: null,
                            start: {
                                offset: 43,
                                line: 2,
                                character: 4,
                            },
                            end: {
                                offset: 57,
                                line: 3,
                                character: 8,
                            },
                        },
                        rvalue: null,
                        comment: null,
                        start: {
                            offset: 43,
                            line: 2,
                            character: 4,
                        },
                        end: {
                            offset: 58,
                            line: 3,
                            character: 9,
                        },
                    },
                    rvalue: null,
                    comment: null,
                    start: {
                        offset: 43,
                        line: 2,
                        character: 4,
                    },
                    end: {
                        offset: 59,
                        line: 3,
                        character: 10,
                    },
                },
                rvalue: null,
                comment: null,
                start: {
                    offset: 43,
                    line: 2,
                    character: 4,
                },
                end: {
                    offset: 60,
                    line: 3,
                    character: 11,
                },
            },
            start: {
                offset: 37,
                line: 1,
                character: 0,
            },
            end: {
                offset: 62,
                line: 4,
                character: 1,
            },
        },
        rvalue: {
            kind: 'FunctionExpression',
            name: 'f',
            params: [
                {
                    kind: 'BinaryExpression',
                    operator: ';',
                    lvalue: {
                        kind: 'Constant',
                        name: 'null',
                        start: {
                            offset: 72,
                            line: 7,
                            character: 4,
                        },
                        end: {
                            offset: 76,
                            line: 7,
                            character: 8,
                        },
                    },
                    rvalue: null,
                    comment: null,
                    start: {
                        offset: 72,
                        line: 7,
                        character: 4,
                    },
                    end: {
                        offset: 77,
                        line: 7,
                        character: 9,
                    },
                },
                {
                    kind: 'BinaryExpression',
                    operator: ';',
                    lvalue: {
                        kind: 'Constant',
                        name: 'null',
                        start: {
                            offset: 83,
                            line: 8,
                            character: 4,
                        },
                        end: {
                            offset: 87,
                            line: 8,
                            character: 8,
                        },
                    },
                    rvalue: null,
                    comment: null,
                    start: {
                        offset: 83,
                        line: 8,
                        character: 4,
                    },
                    end: {
                        offset: 88,
                        line: 8,
                        character: 9,
                    },
                },
                null,
            ],
            start: {
                offset: 65,
                line: 6,
                character: 0,
            },
            end: {
                offset: 91,
                line: 9,
                character: 1,
            },
        },
        comment: null,
        start: {
            offset: 37,
            line: 1,
            character: 0,
        },
        end: {
            offset: 91,
            line: 9,
            character: 1,
        },
    },
    rvalue: null,
    comment: null,
    start: {
        offset: 37,
        line: 1,
        character: 0,
    },
    end: {
        offset: 92,
        line: 9,
        character: 2,
    },
};
