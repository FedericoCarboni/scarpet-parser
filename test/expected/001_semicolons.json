{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "operator": ";",
    "lvalue": {
      "kind": "BinaryExpression",
      "operator": ";",
      "lvalue": {
        "kind": "ParenthesisedExpression",
        "value": {
          "kind": "BinaryExpression",
          "operator": ";",
          "lvalue": {
            "kind": "BinaryExpression",
            "operator": ";",
            "lvalue": {
              "kind": "BinaryExpression",
              "operator": ";",
              "lvalue": {
                "kind": "BinaryExpression",
                "operator": ";",
                "lvalue": {
                  "kind": "Constant",
                  "name": "null",
                  "start": {
                    "pos": 43,
                    "row": 2,
                    "col": 4
                  },
                  "end": {
                    "pos": 47,
                    "row": 2,
                    "col": 8
                  }
                },
                "rvalue": {
                  "kind": "Constant",
                  "name": "null",
                  "start": {
                    "pos": 53,
                    "row": 3,
                    "col": 4
                  },
                  "end": {
                    "pos": 57,
                    "row": 3,
                    "col": 8
                  }
                },
                "comment": null,
                "start": {
                  "pos": 43,
                  "row": 2,
                  "col": 4
                },
                "end": {
                  "pos": 57,
                  "row": 3,
                  "col": 8
                }
              },
              "rvalue": null,
              "comment": null,
              "start": {
                "pos": 43,
                "row": 2,
                "col": 4
              },
              "end": {
                "pos": 58,
                "row": 3,
                "col": 9
              }
            },
            "rvalue": null,
            "comment": null,
            "start": {
              "pos": 43,
              "row": 2,
              "col": 4
            },
            "end": {
              "pos": 59,
              "row": 3,
              "col": 10
            }
          },
          "rvalue": null,
          "comment": null,
          "start": {
            "pos": 43,
            "row": 2,
            "col": 4
          },
          "end": {
            "pos": 60,
            "row": 3,
            "col": 11
          }
        },
        "start": {
          "pos": 37,
          "row": 1,
          "col": 0
        },
        "end": {
          "pos": 62,
          "row": 4,
          "col": 1
        }
      },
      "rvalue": {
        "kind": "FunctionExpression",
        "name": "f",
        "params": [
          {
            "kind": "BinaryExpression",
            "operator": ";",
            "lvalue": {
              "kind": "Constant",
              "name": "null",
              "start": {
                "pos": 72,
                "row": 7,
                "col": 4
              },
              "end": {
                "pos": 76,
                "row": 7,
                "col": 8
              }
            },
            "rvalue": null,
            "comment": null,
            "start": {
              "pos": 72,
              "row": 7,
              "col": 4
            },
            "end": {
              "pos": 77,
              "row": 7,
              "col": 9
            }
          },
          {
            "kind": "BinaryExpression",
            "operator": ";",
            "lvalue": {
              "kind": "Constant",
              "name": "null",
              "start": {
                "pos": 83,
                "row": 8,
                "col": 4
              },
              "end": {
                "pos": 87,
                "row": 8,
                "col": 8
              }
            },
            "rvalue": null,
            "comment": null,
            "start": {
              "pos": 83,
              "row": 8,
              "col": 4
            },
            "end": {
              "pos": 88,
              "row": 8,
              "col": 9
            }
          },
          null
        ],
        "start": {
          "pos": 65,
          "row": 6,
          "col": 0
        },
        "end": {
          "pos": 91,
          "row": 9,
          "col": 1
        }
      },
      "comment": null,
      "start": {
        "pos": 37,
        "row": 1,
        "col": 0
      },
      "end": {
        "pos": 91,
        "row": 9,
        "col": 1
      }
    },
    "rvalue": null,
    "comment": null,
    "start": {
      "pos": 37,
      "row": 1,
      "col": 0
    },
    "end": {
      "pos": 92,
      "row": 9,
      "col": 2
    }
  }
}